docs/book/payments.rst

```diff
@@ -4,112 +4,136 @@
+The available transitions between these states are:
 
-.. code-block:: php
+.. code-block:: yaml
 
-    <?php
+    transitions:
+        create:
+            from: [cart]
+            to: new
+        process:
+            from: [new]
+            to: processing
+        complete:
+            from: [new, processing]
+            to: completed
+        fail:
+            from: [new, processing]
+            to: failed
+        cancel:
+            from: [new, processing]
+            to: cancelled
+        refund:
+            from: [completed]
+            to: refunded
+        void:
+            from: [new, processing]
+            to: void
 
-    echo $payment->getAmount() . $payment->getCurrencyCode();
+Of course, you can define your own states and transitions to create a workflow, that perfectly matches your needs.
+Full configuration can be seen in the `PaymentBundle/Resources/config/app/state_machine.yml <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PaymentBundle/Resources/config/app/state_machine.yml>`_.
 
-    $order = $payment->getOrder(); // Get the order.
+Changes to payment happen through applying appropriate transitions.
 
-    echo $payment->getMethod()->getName(); // Get the name of payment method used.
+How to create a Payment programmatically?
+'''''''''''''''''''''''''''''''''''''''''
 
-Payment State and Workflow
-==========================
+We cannot create a Payment without an Order, therefore let's assume that you have an Order to which you will assign a new payment.
 
-We are using [StateMachine] library to manage all payment states, here is the full list of defaults:
+.. code-block:: php
 
-* new (initial)
-* unknown
-* pending
-* processing
-* completed
-* failed
-* cancelled
-* void
-* refunded
+    $payment = $this->container->get('sylius.factory.payment')->createNew();
 
-Of course, you can define your own states and transitions to create a workflow, that perfectly matches your business. Full configuration can be seen `here <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PaymentBundle/Resources/config/app/state_machine.yml>`_.
+    $payment->setOrder($order);
+    $payment->setCurrencyCode('USD');
 
-Changes to payment happen mostly through applying appropriate transitions.
+    $this->container->get('sylius.repository.payment')->add($payment);
 
 Payment Methods
 ---------------
 
-A [PaymentMethod] represents a way that your customer pays during the checkout process. It holds a reference to specific *gateway* with custom configuration.
-You can have different payment methods using the same gateway, like PayPal or Stripe. Default model of payment method contains the following fields:
+A [PaymentMethod] represents a way that your customer pays during the checkout process. It holds a reference to a specific *gateway* with custom configuration.
+You can have different payment methods using the same gateway, like PayPal or Stripe.
+
+How to create a PaymentMethod programmatically?
```

Comment:
Maybe it would be worthy to change a chapter order, and add a payment gateway section before this one.